search/services/index/api_classic/classic_search.py

```diff
@@ -0,0 +1,49 @@
+"""Translate classic API `Phrase` objects to Elasticsearch DSL."""
+
+from elasticsearch_dsl import Q, Search
+
+from search.domain import ClassicAPIQuery
+from search.services.index.api_classic.query_builder import query_builder
+
+
+def classic_search(search: Search, query: ClassicAPIQuery) -> Search:
+    """
+    Prepare a :class:`.Search` from a :class:`.ClassicAPIQuery`.
+
+    Parameters
+    ----------
+    search : :class:`.Search`
+        An Elasticsearch search in preparation.
+    query : :class:`.ClassicAPIQuery`
+        An query originating from the Classic API.
+
+    Returns
+    -------
+    :class:`.Search`
+        The passed ES search object, updated with specific query parameters
+        that implement the advanced query.
+
+    """
+    # Initialize query.
+    if query.phrase:
+        dsl_query = query_builder(query.phrase)
+    else:
+        dsl_query = Q()
+
+    # Filter id_list if necessary.
+    if query.id_list:
+        # Separate versioned and unversioned papers.
+        paper_ids = [id for id in query.id_list if "v" not in id]
```

Comment:
This unfortunately won't work for a subset of the old-style arXiv identifiers (e.g. `solv-int/9704004`). Something like this should be ok:
```
import re
r1 = re.compile(".*v[1-9]\d*$")
paper_ids = []
paper_id_vs = []
for id in query.id_list:
    paper_id_vs.append(id) if r1.search(id) else paper_ids.append(id)

Reply:
Fixed.